internal/batches/executor/run_steps.go

```diff
@@ -352,13 +197,138 @@ func runSteps(ctx context.Context, opts *executionOpts) (result executionResult,
 	}
 
 	execResult.Diff = string(diffOut)
-	if len(results) > 0 && results[len(results)-1].files != nil {
-		execResult.ChangedFiles = results[len(results)-1].files
-	}
+	execResult.ChangedFiles = previousStepResult.Files
 
 	return execResult, stepResults, err
 }
 
+const workDir = "/work"
```

Comment:
Everything below this line is "just" extracted code from `runSteps`. There is no new logic, it's the same code, but in separate functions.